File: LC263.cpp
class Solution {
public:
    int divide(int a, int b){
        while(a%b == 0)
            a /= b;
        
        return a;
    }

    bool isUgly(int n) {
        if(n <= 0)
            return false;

        for(auto i : {2, 3, 5})
            n = divide(n, i);
        
        return n == 1;
    }
};

File: LC263.py
class Solution:
    def isUgly(self, n: int) -> bool:

        if n <= 0:
            return False
            
        def divide(a, b):
            while a % b == 0:
                a //= b

            return a
        
        for i in (2,3,5):
            n = divide(n, i)
        
        return n == 1
